tutorials/mlflow_queries/tutorial.py

```diff
@@ -0,0 +1,132 @@
+
+import matplotlib
+matplotlib.use('Agg')
+from matplotlib import pyplot as plt
+
+_DESCRIPTION = '''
+A MLFlow+UV tutorial showing how to query results from multiple runs.
+
+Please note that this tutorial requires the matplotlib and numpy packages.
+
+This example will run a simple program and log results to your current directory
+under './mlruns' by default. To view the results, run a local mlflow ui server in the
+same directory:
+
+mlflow ui
+
+By default, this will serve results at http://localhost:5000.
+
+More information on MLFlow tracking can be found at
+https://mlflow.org/docs/latest/tracking.html
+
+This tutorial will also create a plot called 'tutorial.png' of some of its metric
+data in your local directory.
+'''
+
+# simple set of parameters for our runs
+PARAMETERS = [{
+    'mean': 0,
+    'std': 0.1
+}, {
+    'mean': 0,
+    'std': 0.2
+}, {
+    'mean': 1,
+    'std': 0.1
+}]
+
+
+def _gaussian(x: float, mean: float, std: float) -> float:
+  return np.exp(-0.5 * ((x - mean) / std)**2) / (std * np.sqrt(2 * np.pi))
+
+
+def _compute(mean: float, std: float):
+  '''performs a single compute run, logging metrics to mlflow'''
+  fn = lambda x: _gaussian(x, mean, std)
+
+  for i in range(64):
+    x = (i / 32.0) - 1
+    metrics = {'x': x, 'y': fn(x)}
+    uv.report_all(step=i, m=metrics)
+
+
+def _run_experiments(experiment_name: str):
+  '''performs runs for each of our parameter settings, creating the
+  associated mlflow objects and logging paramters'''
+
+  for i, p in enumerate(PARAMETERS):
+    with uv.start_run(
+        experiment_name=experiment_name,
+        run_name=f'run_{i}',
+    ):
+      uv.report_params(p)
+      _compute(**p)
+
+
+def _get_metric_array(client: mlflow.tracking.MlflowClient, run_id: str,
+                      metric: str) -> np.ndarray:
+  '''gets metric data from mlflow and converts it to a numpy array'''
+  return np.array([
+      x.value for x in sorted(client.get_metric_history(run_id, metric),
+                              key=lambda x: x.step)
+  ])
+
+
+def main():
+
+  experiment_name = f'{getpass.getuser()}_query_tutorial'
+
+  with uv.active_reporter(MLFlowReporter()):
+    _run_experiments(experiment_name)
+
+  client = mlflow.tracking.MlflowClient()
+  experiment = client.get_experiment_by_name(experiment_name)
+
+  # get all of our runs where we set mean=0
+  mean_zero_runs = client.search_runs(
+      experiment_ids=[experiment.experiment_id],
+      filter_string='params.mean = "0"',
+  )
+
+  # extract metric data
+  metrics = {}
+  for r in mean_zero_runs:
+    run_id = r.info.run_id
+    run_name = r.data.tags['mlflow.runName']
+    metrics[run_name] = {
+        k: _get_metric_array(client, run_id, k) for k in ('x', 'y')
+    }
+
+  # generate a simple plot and save it to file
+  for k, v in metrics.items():
+    plt.plot(v['x'], v['y'], label=k)
+  plt.legend()
+  plt.grid(True)
+  plt.title('UV/MLFlow Tutorial')
+  plt.savefig('tutorial.png')
```

Comment:
do you want to save it to the actual location that the docs need, so folks could run again to regenerate without moving it?